### PR TITLE
Add package-level documentation

### PR DESCRIPTION
**Pull request by Ironholds**
A fairly common paradigm is to start exploring a new package (or an old one where you can't quuite remember the name of a function) is:

> install.packages("packagename")
> library(packagename)
> ?packagename

This gets you trivial access to the package index, which is super nice. However, it falls over if the package doesn't have documentation at the package level. This patch adds that documentation for RMySQL, albeit as a slightly-cleaned-up version of the relevant DESCRIPTION entries.

### CODE REVIEW

**Comment by Ironholds** [comment]
Wait, nope, I'm just a dolt who didn't spot there was already (if not package level documentation, then) a redirect. Ignore this.

**Comment by krlmlr** [comment]
Actually, this looks nice to me. Why did you close?

File: R/RMySQL.R [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#' @title RMySQL: Database Interface and MySQL Driver for R
+#' @name RMySQL
+#' @description Implements a DBI interface to MySQL and MariaDB databases.
+#'
+#' @seealso the \href{https://cran.r-project.org/web/packages/RMySQL/RMySQL.pdf}{package documentation}.
+#' @useDynLib RMySQL
+#' @importFrom Rcpp sourceCpp
+#' @docType package
+#' @aliases RMySQL RMySQL-package
+NULL
```
Comment (krlmlr) [inline R/RMySQL.R]: You can write `"_PACKAGE"` instead of `NULL` and get rid of a few tags above.

**Comment by Ironholds** [comment]
@krlmlr I mean, the thing I was trying to correct was an absence of ?RMySQL working, and it now does! Admittedly this is somewhat nicer than dropping people in a class definition. Up to you.

File: R/RMySQL.R (hunk 1 above)
Comment (Ironholds) [inline R/RMySQL.R]: So "_RMySQL"? Which tags? (I've always gone for the NULL approach)

Comment (krlmlr) [inline R/RMySQL.R]: No, literally `"_PACKAGE"`. Currently, title, description, and doctype are autogenerated (drawn from DESCRIPTION), everything else is still supported.

**Comment by krlmlr** [comment]
?RMySQL doesn't work on my system (but ?MySQL does), which version of RMySQL do you have?

**Comment by Ironholds** [comment]
0.10.9 at this end; you?

**Comment by krlmlr** [comment]
cb49a91dd33e71616d3cbca713b39a2f5d8381aa ;-)

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**